Switch page to markdown/mkdocs

vvolkl opened the pull request:
A test deployment is available under [cvmfs.docs.cern.ch](cvmfs.docs.cern.ch). I'm planning to use that as the main URL at some point - cvmfs.readthedocs.io can either mirror or redirect. It shouldn't be a problem either way as readthedocs.io [does support mkdocs](https://docs.readthedocs.com/platform/stable/intro/mkdocs.html).

* The biggest issue in keeping the site the same is that mkdocs has no native/simple way to handle different versions of the documentation. There's https://github.com/jimporter/mike but I'm wondering if it isn't actually better to stop building separate versions, and add colored boxes for "New in version... " and "Deprecated in ..." where deemed necessary.

* RST does have a few advantages, in particular the citations have a better syntax. There's a mkdocs-bibtex plugin, but I thought it overkill, plus I couldn't get a full list of references to render correctly. So the citations are basically manually linked to the anchors in the list at the end.

- [ ] I've started reorganising the navigation a bit, but it's not complete yet. The links shouldn't be affected as those point to the filenames, and those did not change.
- [ ] There were quite a few conversion issues (internal links, tables, ...) and I haven't caught them all. The whole documentation should be checked.
- [ ] The .rst files can be removed, and the mkdocs subfolder contents moved up

vvolkl committed:
mkdocs

vvolkl committed:
up

vvolkl committed:
add logo

DrDaveD commented:
>  I'm wondering if it isn't actually better to stop building separate versions, and add colored boxes for "New in version... " and "Deprecated in ..." where deemed necessary.

Please don't do that.  Separate versions of the documentation is much preferable.  Surely there must be a way to do it using different URLs generated from different git branches, even if it isn't native/simple.

> A test deployment is available under https://cvmfs.docs.cern.ch

A quick check of the Appendix -> parameters shows problems there.  Probably corresponds to one of the conversion issues.

vvolkl commented:
> Please don't do that. Separate versions of the documentation is much preferable. Surely there must be a way to do it using different URLs generated from different git branches, even if it isn't native/simple.

Looking again, it is actually fairly simple to do this in mkdocs+mike, it'll put different versions under cvmfs.docs.cern.ch/<version> as usual, but the version selector seems to be buggy. However, that can be worked around for now with putting the links to the available versions on the front page

vvolkl committed:
add tabbed extension

vvolkl committed:
fix main url

vvolkl committed:
use content tabs for quickstart, part 1

vvolkl commented:
It seems like mkdocs is no longer maintained and [zensical](zensical.org) will replace it. Zensical does not support the readthedocs theme or the version switcher at them moment however, so it's probably better to stick with mkdocs for now. Moving from mkdocs to zensical should be easy, once everything is converted to markdown.

vvolkl committed:
style: add tabbed extension styling and fix tab syntax

Agent-Id: agent-c29c8d5c-2312-4be5-aa09-279cdcebe5e0
Linked-Note-Id: 458505b6-f28f-4122-a4de-fa5d0261e815

vvolkl committed:
docs: migrate from Sphinx/RST to MkDocs/Markdown format

Agent-Id: agent-3bbcf57c-48a0-4f88-ad38-e735f1faeb78

vvolkl committed:
up

vvolkl committed:
up

vvolkl committed:
up

vvolkl committed:
up

vvolkl committed:
up

vvolkl committed:
internal links

vvolkl committed:
parameter list

vvolkl committed:
up

vvolkl committed:
up

vvolkl committed:
docs: fix parameter descriptions formatting and clarity

Agent-Id: agent-d8b56863-51db-4bb9-976a-3084639e44f4
Linked-Note-Id: 928880a0-b4a0-4c22-b692-83a449e6d0c6

vvolkl committed:
docs: convert definition lists to structured format and style repository

Agent-Id: agent-1627e044-f3be-486f-bc42-330333e14c9b
Linked-Note-Id: 2717f4f7-8d5a-48f5-9c56-ba2bf0d4afdd

christge-sft committed:
Convert all tables in apx to markdown definitions lists and provide a nice desplay with css. (#259)

* Convert the Local Backend Storage Infrastructure subsection to a definition list, and render it as a three column table

* Convert the Server Spool Area of a Repo subsection to a definition list

* Convert the Repo Config Dir subsection to a definition list

* Apply the same css sugar for the definition lists of apx available packages

* Update docs/_static/css/custom.css

---------

Co-authored-by: Valentin Volkl <valentin.volkl@cern.ch>

vvolkl commented:
It's now also possible to preview this on readthedocs: https://cvmfs.readthedocs.io/en/mkdocs/

There may be a few smaller issues with lists and internal links remaining, so still need to do a proofread, but I think this is nearing completion.

vvolkl committed:
consistent uppercase NFS

vvolkl committed:
definition list and code blocks improvement

vvolkl committed:
better code blocks

vvolkl committed:
nicer styling of /cvmfs

vvolkl committed:
fix fqrn syntax

vvolkl committed:
update author list

vvolkl committed:
final fixes
